### PR TITLE
Issues/5688: copy all modified waypoints to notes

### PR DESCRIPTION
**Pull request by eddiemuc**
This PR attempts to solve Issue #5688: https://github.com/cgeo/cgeo/issues/5688

The following "Waypoint text pattern" (for both creating and parsing) was created:
@name (type) coordinate\n"user note"

Example:
@my Waypoint (puzzle) N48 10.123 E11 05.456
"My User Note"

Additional notes:
- Regarding the @: if no @ is found during parse then no name is created from text (default name applies). If there would not be a name marker, then coords parsed from within text might get very strange names...
- Type: type will also be recognized when NOT in () like today, but will then additionally be part of the name. E.g. "@parking Spot N48 10.123 E11 05.456" will become waypoint of type "Parking" with name "Parking spot". Type in () will be filtered out in name creation.
- Type: for easier typing I created one-char-shortcuts for each waypoint type: F=final, P=parking, X=puzzle,... E.g. "@the End (F) N48 10.123 E11 05.456" will be parsed as Final waypoint
- User note: " can be used optionally because user note can be multilined. Escaped " in user note (") are recognized. " to start note can be placed both on same line (after coord) and on next line. If no " is found after coord, then "user note" is filled from coord until next linebreak.

personalnote
waypoints

### CODE REVIEW

**Comment by Lineflyer** [comment]
ok to test

**Comment by Lineflyer** [comment]
retest this please

**Comment by eddiemuc** [comment]
Looks like an issue with some test with connectivity issues from time to time (cgeo.geocaching.address.GeocoderTest.testAndroidGeocoder). Seems to happen from time to time on Jenkins.
Is there something I can do here?

**Comment by eddiemuc** [comment]
Retest this please

**Comment by Lineflyer** [comment]
We did not find out yet, why the geocoder test is failing randomly. Pretty sure not related to your PR. Nothing you can do about it, unless you find out the reason for failing ;)

**Comment by Lineflyer** [comment]
As we currently discuss to cut off a new release from master, may I suggest to merge this new feature PR only after that?

**Comment by eddiemuc** [comment]
Sure this seems very reasonable

**Comment by moving-bits** [comment]
> > As we currently discuss to cut off a new release from master, may I suggest to merge this new feature PR only after that?
>
> Sure this seems very reasonable

Therefore setting the "Do not merge" label for now.

**Comment by moving-bits** [comment]
@eddiemuc 
As we have conflicting changes meanwhile: Can you rebase and squash your PR? Thx.

**Comment by eddiemuc** [comment]
will do this in the next couple of days, I hope that's soon enough...

**Comment by eddiemuc** [comment]
Wow this looks messy. Seems like I followed the wrong internet page for squashing and rebasing instructions.

File: main/src/cgeo/geocaching/CacheDetailActivity.java [hunk 1]
```diff
@@ -184,6 +184,8 @@
 
     private static final String EXTRA_FORCE_WAYPOINTSPAGE = "cgeo.geocaching.extra.cachedetail.forceWaypointsPage";
 
+    //private static final int MAX_USERNOTE_SIZE = 2400; //max. allowed size on Geocaching.com
```
[the remark this answers is not in the document]
Comment (eddiemuc) [reply main/src/cgeo/geocaching/CacheDetailActivity.java]: done

**Comment by eddiemuc** [comment]
OK I took some lessons in git and learned a lot about interactive rebase. PR should be clean now, I also resolved @moving-bits 's comment

**Comment by moving-bits** [comment]
Then let's merge it :-)
Thanks @eddiemuc for your contribution!

**Pull request merged**